src/filesystem/implementation.js

```diff
@@ -1678,6 +1678,25 @@ function mkdir(fs, context, path, mode, callback) {
   make_directory(context, path, callback);
 }
 
+function mkdtemp(fs, context, prefix, options, callback) { 
+  callback = arguments[arguments.length - 1];
+  let shared = require('../shared.js');
+  if(!prefix) {
+    return callback(new Error('filename prefix is required'), prefix);
+  } else {
+    let random = shared.randomChars(6);
+    var path = prefix+'-'+random; 
+    normalize(prefix);
+    make_directory(context, path, function(error) {
+      if (error) {
```

Comment:
Simplify this to: `callback(error, path);` (it will be `undefined` if not set, so you don't need to bother with the `if` checks.